examples/scenarioConstrainedDynamics.py

```diff
@@ -0,0 +1,277 @@
+
+    # Create the simulation process and specify the integration update time
+    simulationTimeStep = macros.sec2nano(0.01)  # update process rate update time
+    dynProcess = scSim.CreateNewProcess(simProcessName)
+    dynProcess.addTask(scSim.CreateNewTask(simTaskName, simulationTimeStep))
+
+    print(simulationTimeStep)
+
+    # Create both spacecraft
+    scObject1 = spacecraft.Spacecraft()
+    scObject1.ModelTag = "spacecraftBody1"
+    scObject2 = spacecraft.Spacecraft()
+    scObject2.ModelTag = "spacecraftBody2"
+
+    # Set the integrator to RKF45
+    integratorObject1 = svIntegrators.svIntegratorRKF45(scObject1)
+    scObject1.setIntegrator(integratorObject1)
+    # Sync dynamics integration across both spacecraft
+    scObject1.syncDynamicsIntegration(scObject2)
+
+    # Define mass properties of the rigid hub of both spacecraft
+    scObject1.hub.mHub = 750.0
+    scObject1.hub.r_BcB_B = [[0.0], [0.0], [1.0]]
+    scObject1.hub.IHubPntBc_B = [[600.0, 0.0, 0.0], [0.0, 600.0, 0.0], [0.0, 0.0, 600.0]]
+    scObject2.hub.mHub = 750.0
+    scObject2.hub.r_BcB_B = [[0.0], [0.0], [1.0]]
+    scObject2.hub.IHubPntBc_B = [[600.0, 0.0, 0.0], [0.0, 600.0, 0.0], [0.0, 0.0, 600.0]]
+
+    # Add Earth gravity to the simulation if requested
+    if env == 'Gravity':
+        earthGravBody = gravityEffector.GravBodyData()
+        earthGravBody.planetName = "earth_planet_data"
+        earthGravBody.mu = 0.3986004415E+15  # [meters^3/s^2]
+        earthGravBody.isCentralBody = True
+        scObject1.gravField.gravBodies = spacecraft.GravBodyVector([earthGravBody])
+        scObject2.gravField.gravBodies = spacecraft.GravBodyVector([earthGravBody])
+        # set orbital altitude to LEO
+        oe = orbitalMotion.ClassicElements()
+        oe.a = earthGravBody.radEquator + 7500e3  # meters
+        oe.e = 0.01
+        oe.i = 30.0 * macros.D2R
+        oe.Omega = 60.0 * macros.D2R
+        oe.omega = 15.0 * macros.D2R
+        oe.f = 90.0 * macros.D2R
+        r_B2N_N, rDot_B2N_N = orbitalMotion.elem2rv(earthGravBody.mu, oe)
+    else: # If no gravity requested, place in free-floating space
+        r_B2N_N = np.array([1,1,1])
+        rDot_B2N_N = np.array([1,1,1])
+
+    # Set initial attitudes to zero (B1, B2, and N frames all initially aligned)
+    dir = r_B2N_N/np.linalg.norm(r_B2N_N)
+    l = 0.1
+    COMoffset = 0.1 # distance from COM to where the arm connects to the spacecraft hub, same for both spacecraft [meters]
+    r_P1B1_B1 = np.dot(dir,COMoffset)
+    r_P2B2_B2 = np.dot(-dir,COMoffset)
+    r_P2P1_B1Init = np.dot(dir,l)
+    r_B1N_N = r_B2N_N + r_P2B2_B2 - r_P2P1_B1Init - r_P1B1_B1
+    rDot_B1N_N = rDot_B2N_N
+
+    # Compute rotational states
+    # let C be the frame at the combined COM of the two vehicles
+    r_CN_N = (r_B1N_N * scObject1.hub.mHub + r_B2N_N * scObject2.hub.mHub) / (scObject1.hub.mHub + scObject2.hub.mHub)
+    r_B1C_N = r_B1N_N - r_CN_N
+    r_B2C_N = r_B2N_N - r_CN_N
```

Comment:
unused variable, please remove

Reply:
These offsets are used in computing the relative velocities for the scenario, so they are required.